### PR TITLE
mgr/dashboard: migrated config table tab to resource page

### PR DESCRIPTION
**Pull request by syedali237**
- Fixes: https://tracker.ceph.com/issues/78963

https://github.com/user-attachments/assets/9db5e287-72c1-425a-bf8c-00308fd944d0

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins test classic perf` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-perf-classic/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml)
- `jenkins test crimson perf` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-perf-crimson/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml)
- `jenkins test signed` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pr-commits/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-commits/config/definitions/ceph-pr-commits.yml)
- `jenkins test make check` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pull-requests/config/definitions/ceph-pull-requests.yml)
- `jenkins test make check arm64` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pull-requests-arm64/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pull-requests-arm64/config/definitions/ceph-pull-requests-arm64.yml)
- `jenkins test submodules` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-pr-submodules/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-submodules/config/definitions/ceph-pr-commits.yml)
- `jenkins test dashboard` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-dashboard-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-dashboard-pull-requests/config/definitions/ceph-dashboard-pull-requests.yml)
- `jenkins test dashboard cephadm` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-dashboard-cephadm-e2e/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-dashboard-cephadm-e2e/config/definitions/ceph-dashboard-cephadm-e2e.yml)
- `jenkins test api` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-api/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-api/config/definitions/ceph-pr-api.yml)
- `jenkins test docs` [ReadTheDocs](https://readthedocs.org/projects/ceph/) | [Github Workflow Definition](https://github.com/ceph/ceph/blob/main/.readthedocs.yml)
- `jenkins test ceph-volume all` [Jenkins Jobs](https://jenkins.ceph.com/view/ceph-volume%20PR/) | [Jenkins Jobs Definition](https://github.com/ceph/ceph-build/blob/main/ceph-volume-cephadm-prs/config/definitions/ceph-volume-pr.yml)
- `jenkins test windows` [Jenkins Job](https://jenkins.ceph.com/job/ceph-windows-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-windows-pull-requests/config/definitions/ceph-windows-pull-requests.yml)
- `jenkins test rook e2e` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-orchestrator-rook-e2e/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-rook-e2e/config/definitions/ceph-orchestrator-rook-e2e.yml)

You must only issue one Jenkins command per-comment. Jenkins does not understand
comments with more than one command.
</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration-resource-sidebar/configuration-resource-sidebar.component.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import { ConfigurationResourceStateService } from '~/app/shared/services/configuration-resource-state.service';
+
+@Component({
+  selector: 'cd-configuration-resource-sidebar',
+  templateUrl: './configuration-resource-sidebar.component.html',
+  styleUrls: ['./configuration-resource-sidebar.component.scss'],
+  providers: [ConfigurationResourceStateService],
+  standalone: false
+})
+export class ConfigurationResourceSidebarComponent implements OnInit, OnDestroy {
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration-resource-sidebar/configuration-resource-sidebar.component.ts:15]: Sidebar spec file is empty (0 bytes). Add tests: paramMap subscription, sidebar items construction, decode URI edge cases. The mgr-module sidebar spec in #70764 is a good reference.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration-resource-sidebar/configuration-resource-sidebar.component.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    this.sidebarItems = [
+      {
+        label: $localize`Overview`,
+        route: [this.basePath, this.configurationOption, 'overview'],
+        routerLinkActiveOptions: { exact: true }
+      }
+    ];
+  }
+
+  constructor(
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration-resource-sidebar/configuration-resource-sidebar.component.ts]: Move constructor before `ngOnInit`.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration-resource-page/configuration-resource-breadcrumb.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ConfigurationResourceBreadcrumbResolver extends BreadcrumbsResolver {
+  resolve(route: ActivatedRouteSnapshot): IBreadcrumb[] {
+    const configOption = this.getConfigOptionName(route.paramMap.get('name') || '');
+
+    return [
+      { text: 'Administration/Configuration', path: '/configuration' },
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration-resource-page/configuration-resource-breadcrumb.resolver.ts]: `'Administration/Configuration'` is user-visible -- mark with `$localize`.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.ts [hunk 4]
```diff
@@ -161,7 +166,10 @@ export class ConfigurationComponent extends ListWithDetails implements OnInit {
   getConfigurationList(context: CdTableFetchDataContext) {
     this.configurationService.getConfigData().subscribe(
       (data: any) => {
-        this.data = data;
+        this.data = (Array.isArray(data) ? data : []).map((configOption: any) => ({
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.ts]: Type this instead of `any`. Use the `ConfigurationOption` type from `configuration-resource-state.service`.

**Review by afreen23 — CHANGES_REQUESTED** [review]

**Review by afreen23 — APPROVED** [review]
LGTM

**Comment by syedali237** [comment]
jenkins test make check

**Comment by syedali237** [comment]
jenkins test make check arm64

**Comment by syedali237** [comment]
jenkins test dashboard

**Comment by syedali237** [comment]
jenkins test windows

**Comment by syedali237** [comment]
jenkins test api

**Comment by syedali237** [comment]
jenkins test api

**Comment by syedali237** [comment]
jenkins test make check

**Comment by syedali237** [comment]
jenkins test  make check

**Review by afreen23 — CHANGES_REQUESTED** [review]
Image

When only one tab no need to show Overview

**Comment by syedali237** [comment]
jenkins test dashboard

File: src/pybind/mgr/dashboard/frontend/src/app/shared/services/configuration-resource-state.service.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  readonly configuration$ = this.configurationSource.asObservable();
+
+  constructor(private configurationService: ConfigurationService) {}
+
+  load(configNameRoute: string): void {
+    if (!configNameRoute) {
+      this.configurationSource.next(null);
+      return;
+    }
+
```
Comment (afreen23) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/services/configuration-resource-state.service.ts:43]: same issue as in `service-resource-state.service.ts` — each `load()` call creates an untracked subscription. Use `switchMap` or store and unsubscribe before starting a new call to prevent race conditions when route params change quickly

**Review by afreen23 — CHANGES_REQUESTED** [review]

**Comment by syedali237** [comment]
jenkins test windows

**Review by afreen23 — APPROVED** [review]
